cmd/oras/root/manifest/index/create.go

```diff
@@ -0,0 +1,186 @@
+		Short: "[Experimental] Create and push an index from provided manifests",
+		Long: `[Experimental] Create and push an index from provided manifests. All manifests should be in the same repository
+
+Example - create an index from source manifests tagged 'linux-amd64' and 'linux-arm64', and push without tagging:
+  oras manifest index create localhost:5000/hello linux-amd64 linux-arm64
+
+Example - create an index from source manifests tagged 'linux-amd64' and 'linux-arm64', and push with the tag 'latest':
+  oras manifest index create localhost:5000/hello:latest linux-amd64 linux-arm64
+
+Example - create an index from source manifests using both tags and digests, and push with tag 'latest':
+  oras manifest index create localhost:5000/hello:latest linux-amd64 sha256:xxx
+
+Example - create an index and push it with multiple tags:
+  oras manifest index create localhost:5000/hello:tag1,tag2,tag3 linux-amd64 linux-arm64 sha256:xxx
+`,
+		Args: oerrors.CheckArgs(argument.AtLeast(1), "the destination index to create."),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			refs := strings.Split(args[0], ",")
+			opts.RawReference = refs[0]
+			opts.extraRefs = refs[1:]
+			opts.sources = args[1:]
+			if err := option.Parse(cmd, &opts); err != nil {
+				return err
+			}
+			// check that all tags given by the user are valid tags
+			var allRefs []string
+			if opts.Reference != "" {
+				allRefs = append(opts.extraRefs, opts.Reference)
+			} else {
+				allRefs = opts.extraRefs
+			}
+			for _, ref := range allRefs {
+				if !contentutil.IsValidTag(ref) {
+					return fmt.Errorf("%s is not a valid tag", ref)
+				}
+			}
+			return nil
+		},
+		Aliases: []string{"pack"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createIndex(cmd, opts)
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	return oerrors.Command(cmd, &opts.Target)
+}
+
+func createIndex(cmd *cobra.Command, opts createOptions) error {
+	ctx, logger := command.GetLogger(cmd, &opts.Common)
+	target, err := opts.NewTarget(opts.Common, logger)
+	if err != nil {
+		return err
+	}
+	manifests, err := resolveSourceManifests(ctx, target, opts)
+	if err != nil {
+		return err
+	}
+	index := ocispec.Index{
+		Versioned: specs.Versioned{
+			SchemaVersion: 2,
+		},
+		MediaType: ocispec.MediaTypeImageIndex,
+		Manifests: manifests,
+	}
+	indexBytes, _ := json.Marshal(index)
+	desc := content.NewDescriptorFromBytes(ocispec.MediaTypeImageIndex, indexBytes)
+	opts.Printf("%s  %s  %s\n", status.IndexPromptPacked, descriptor.ShortDigest(desc), ocispec.MediaTypeImageIndex)
+	return pushIndex(ctx, target, desc, indexBytes, opts.Reference, opts.extraRefs, opts.AnnotatedReference(), opts.Printer)
+}
+
+func resolveSourceManifests(ctx context.Context, target oras.ReadOnlyTarget, opts createOptions) ([]ocispec.Descriptor, error) {
+	resolved := []ocispec.Descriptor{}
+	for _, source := range opts.sources {
+		opts.Printf("%s  %s\n", status.IndexPromptFetching, source)
```

Comment:
No need to add extra space.
```suggestion
		opts.Println(status.IndexPromptFetching, source)
```